lua/conform/runner.lua

```diff
@@ -375,8 +375,14 @@ local function run_formatter(bufnr, formatter, config, ctx, input_lines, opts, c
     uv.fs_write(fd, buffer_text)
     uv.fs_close(fd)
     callback = util.wrap_callback(callback, function()
+      local rel_dirname = vim.fs.dirname(vim.fs.relpath(ctx.dirname, ctx.filename))
       log.debug("Cleaning up temp file %s", ctx.filename)
       uv.fs_unlink(ctx.filename)
+      if rel_dirname ~= "." then
+        local dirname = vim.fs.joinpath(ctx.dirname, rel_dirname)
+        log.debug("Cleaning up temp dir %s", dirname)
+        uv.fs_rmdir(dirname)
+      end
```

Comment:
If we're going to go to the trouble to removing dangling directories, I'd like to make sure it's a thorough solution rather than a special case for if it's a single directory deep. Since we're creating the directories just above this, it would be pretty easy to do a quick check there to keep track of which ones in that path already exist, and then we can attempt to remove the set of new directories on the way out.